Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -35,1657 +35,1664 @@
+                        //  https://standards.buildingsmart.org/IFC/RELEASE/IFC4_1/FINAL/HTML/link/ifclocalplacement.htm
+                        // ENTITY IfcLocalPlacement
+                        //	SUBTYPE OF(IfcObjectPlacement);
+                        //		PlacementRelTo: OPTIONAL IfcObjectPlacement; // was promoted to ObjectPlacement from IFC4x2+
+                        //		RelativePlacement: IfcAxis2Placement;
+                        //	WHERE
+                        //		WR21 : IfcCorrectLocalPlacement(RelativePlacement, PlacementRelTo);
+                        // END_ENTITY;
+                        // **************************************************************************************************************************
+
+                        // RelativePlacement				
+                        // TYPE IfcAxis2Placement = SELECT (
+                        //	IfcAxis2Placement2D,
+                        //	IfcAxis2Placement3D);
+                        // END_TYPE;
+                        convertRelativePlacement(object_placement_matrix, local_placement);
+                        carve::math::Matrix relative_placement(carve::math::Matrix::IDENT());
+                        convertIfcLocalPlacementRelationPoint(alreadyApplied, local_placement, relative_placement);
+                        object_placement_matrix = relative_placement * object_placement_matrix;
+
+                    }
+
+                    void convertRelativePlacementOriginInIfcLinearPlacement(std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcObjectPlacement>>& alreadyApplied, carve::math::Matrix& object_placement_matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement> linear_placement);
+
+                    void checkLinearPlacementAgainstAbsolutePlacement(carve::math::Matrix& matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement> linear_placement)
+                    {
+                        // ***********************************************************
+                        // check with the provided CartesianPosition
+                        // CartesianPosition OPTIONAL
+                        carve::math::Matrix absolute_placement(carve::math::Matrix::IDENT());
+                        if(linear_placement->CartesianPosition) {
+                            convertIfcAxis2Placement3D(linear_placement->CartesianPosition.get().lock(), absolute_placement);
+                            if(absolute_placement != matrix) {
+                                BLUE_LOG(trace) << linear_placement->getErrorLog() << "Absolute placemet and the calculated linear placement do not agree";
+                            }
+                        }
+                    }
+
+                    carve::geom::vector<3> getOffsetFromCurve(double length_factor, std::shared_ptr<typename IfcEntityTypesT::IfcDistanceExpression> distExpr)

```

Comment:
Doxycomment missing. Also, the name of the function is not really reflecting what it does.

Reply:
Done